core/src/main/java/com/opensymphony/xwork2/DefaultActionInvocation.java

```diff
@@ -508,6 +509,13 @@ protected String saveResult(ActionConfig actionConfig, Object methodResult) {
     public ActionInvocation serialize() {
         DefaultActionInvocation that = this;
         that.container = null;
+
+        if(that.getInvocationContext() != null && that.getInvocationContext().getContextMap() != null) {
+            Map<String, Object> thatContextMap = that.getInvocationContext().getContextMap();
+            thatContextMap.remove(ServletActionContext.HTTP_REQUEST);
+            thatContextMap.remove(ServletActionContext.HTTP_RESPONSE);
```

Comment:
I'm not 100% sure but in such way you will break Token Session support as I far remember it will try to restore previous request on invalid token.

Reply:
No, as I see, currently, it restores previous `invocation` and replaces previous `request` and `response` with current ones. It does not use them.

```java
            ActionInvocation savedInvocation = InvocationSessionStore.loadInvocation(sessionTokenName, token);

            if (savedInvocation != null) {
                // set the valuestack to the request scope
                ValueStack stack = savedInvocation.getStack();
                request.setAttribute(ServletActionContext.STRUTS_VALUESTACK_KEY, stack);//current request

                ActionContext savedContext = savedInvocation.getInvocationContext();
                savedContext.getContextMap().put(ServletActionContext.HTTP_REQUEST, request);//replace prev req with current
                savedContext.getContextMap().put(ServletActionContext.HTTP_RESPONSE, response);//replace prev res with current
                Result result = savedInvocation.getResult();

                if ((result != null) && (savedInvocation.getProxy().getExecuteResult())) {
                    result.execute(savedInvocation);
                }

                // turn off execution of this invocations result
                invocation.getProxy().setExecuteResult(false);

                return savedInvocation.getResultCode();//no usage of prev req

```

Reply:
`InvocationSessionStore` also does not use previous `request` and use session to restore previous `invocation`.